pkg/apis/serving/v1beta1/revision_defaults.go

```diff
@@ -76,6 +76,7 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 				rs.PodSpec.Containers[idx].Resources.Limits[corev1.ResourceMemory] = *rsrc
 			}
 		}
+
```

Comment:
Maybe fetch a pointer to the probe once to make this a bit more readable.

```
readinessProbe := &rs.PodSpec.Containers[idx].ReadinessProbe
```

I think that works, feel free to ignore if it doesn't.

Reply:
Yes, it worked and it makes more readable! Thank you for the suggestion. Updated.

Reply:
Ugh, but now you need to derefence on every call. May I instead suggest you obtain a pointer to the container at the start of the loop?

```
ctr := &rs.PodSpec.Containers[idx]
```

You can then replace a lot of boilerplate in the whole function and simplify all of the below to `ctr.ReadinessProbe...` which is simple enough I think. No dereferences should be needed then.

Reply:
Thank you! Updated. (Sorry, I should have noticed it by myself.)